lib/presentation/controller/nutritional_value_controller.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:get/get.dart';
+
+import '../../domain/usecase/i_usecase.dart';
+import '../../domain/usecase/implementations/get_nutritional_value_usecase.dart';
+
+class NutritionalValueController extends GetxController
+    with StateMixin<String> {
+  final UseCase getNutritionalValueUseCase =
+      Get.find<GetNutritionalValueUseCase>();
+
+  int? recipeId;
+
+  Future<void>? getNutritionalValue(int? id) async {
+    change(
+      null,
+      status: RxStatus.loading(),
+    );
+
+    if (id != null) {
+      recipeId = id;
+    }
```

Comment:
nutritional Value could be null?

Reply:
That attribute is now used in the onReady function. If that function is called and id is still null it's because the details page was accessed directly either by refreshing the page or manually inputting the URL.